drake_ros_tf2/src/utilities/name_conventions.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "drake_ros_tf2/utilities/name_conventions.h"
+
+#include <sstream>
+#include <string>
+#include <unordered_set>
+
+namespace drake_ros_tf2 {
+namespace utilities {
+
+namespace {
+
+std::string ReplaceAllOccurrences(std::string string, const std::string& target,
+                                  const std::string& replacement) {
+  std::string::size_type n = 0;
+  while ((n = string.find(target, n)) != std::string::npos) {
+    string.replace(n, target.size(), replacement);
+    n += replacement.size();
+  }
+  return string;
+}
+
+}  // namespace
+
+std::string GetTfFrameName(
```

Comment:
https://github.com/RobotLocomotion/drake-ros/pull/35/commits/272cbe56a28e6ece84f14517d4a27be7f318f2ab

I separated the formulation of the names into `CalcTfFrameName` but left it out of the API, to prevent any breakage when things change as we continue working on the integration. Let me know what you think.

Tests were also added.